operator-pipeline-images/operatorcert/entrypoints/detect_changed_operators.py

```diff
@@ -25,36 +26,104 @@ def setup_argparser() -> argparse.ArgumentParser:
     )
     parser.add_argument(
         "--repo-path",
+        type=pathlib.Path,
         help="Path to the root of the local clone of the repo",
         required=True,
     )
     parser.add_argument(
-        "--head-commit", help="Commit ID for the head of the PR", required=True
+        "--base-repo-path",
+        type=pathlib.Path,
+        help="Path to the root of the local clone of the base branch of the PR",
+        required=True,
     )
     parser.add_argument(
-        "--base-commit", help="Commit ID for the base of the PR", required=True
+        "--pr-url",
+        type=str,
+        help="URL of the corresponding pull request",
+        required=True,
     )
     parser.add_argument(
-        "--output-file", help="Path to a json file where results will be stored"
+        "--output-file",
+        type=pathlib.Path,
+        help="Path to a json file where results will be stored",
     )
     parser.add_argument("--verbose", action="store_true", help="Verbose output")
 
     return parser
 
 
-def detect_changed_operators(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements
-    repo_path: str,
-    head_commit: str,
-    base_commit: str,
+def github_pr_affected_files(pr_url: str) -> set[str]:
```

Comment:
There is already a function in `__init__.py` `get_files_added_in_pr`. Please consider reusing it or at least replacing it with your solution.

Reply:
I'm aware of that function, but it contains more logic than a simple "get me a list of affected files" and it's used in 10 different spots across the code base. I think that such a refactoring is way out of scope for this PR.

Reply:
ok, make sense